.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -733,6 +733,28 @@ private DateTimeResolutionResult ParseDuration(string text, DateObject reference
                         beginDate = referenceDate.AddDays(1);
                         endDate = DurationParsingUtil.ShiftDateTime(durationResult.Timex, beginDate, true);
                     }
+                    else
+                    {
+                        var suffixMatch = config.FutureRegex.Match(afterStr);
+                        if (suffixMatch.Success)
+                        {
+                            mod = TimeTypeConstants.afterMod;
+
+                            // For future the beginDate should add 1 first
+                            beginDate = referenceDate.AddDays(1);
+                            endDate = DurationParsingUtil.ShiftDateTime(durationResult.Timex, beginDate, true);
+                        }
+                    }
+
+                    var futureSuffixMatch = config.FutureSuffixRegex.Match(afterStr);
+                    if (futureSuffixMatch.Success)
+                    {
+                        mod = TimeTypeConstants.afterMod;
+
+                        // For future the beginDate should add 1 first
+                        beginDate = referenceDate.AddDays(1);
+                        endDate = DurationParsingUtil.ShiftDateTime(durationResult.Timex, beginDate, true);
```

Comment:
Does it make sense for the body of this if to be a method? It's repeated above too.